crypto/conf/conf_lib.c

```diff
@@ -292,13 +305,16 @@ int NCONF_get_number_e(const CONF *conf, const char *group, const char *name,
     if (str == NULL)
         return 0;
 
-    if (conf == NULL)
-        *result = strtol(str, &str, 10);
-    else
-        for (*result = 0; conf->meth->is_number(conf, *str);) {
-            *result = (*result) * 10 + conf->meth->to_int(conf, *str);
-            str++;
-        }
+    if (conf != NULL) {
+        if (conf->meth->is_number != NULL)
+            is_number = conf->meth->is_number;
+        if (conf->meth->to_int != NULL)
+            to_int = conf->meth->to_int;
+    }
+    for (*result = 0; is_number(conf, *str);) {
+        *result = (*result) * 10 + to_int(conf, *str);
+        str++;
+    }
```

Comment:
Is tracking carry *that* complicated? Given unsigned long val and temp
```
temp = 10 * val + to_int(conf, *str);
if (temp < val) {
    val = BAD; break;
} else {
    val = temp;
}
```
As for saturation one can wonder if it would be appropriate to push error with CONFerr and return error condition.

On side note here is another problem with code path in question. CONF_get_number reads
```
if (status == 0) {
    /* This function does not believe in errors... */
    ERR_clear_error();
}
```
Isn't it kind of overstatement? One can probably say that "this function does not believe in **callee's** errors", but not everything. So ERR_set_mark/ERR_pop_to_mark?

Reply:
> temp = 10 * val + to_int(conf, *str);
>     if (temp < val) {

Obviously insufficient. I mean this is not guaranteed catch overflow. Apologies for potential confusion.

Reply:
> On side note here is another problem with code path in question. CONF_get_number reads
```
if (status == 0) {
    /* This function does not believe in errors... */
    ERR_clear_error();
}
```

Yeah, I'm not sure what to do here.  It kind of makes raising an error for the overflow a problem too.  Presumably there was a reason for this.

Reply:
>> CONF_get_number reads .. ERR_clear_error
> I'm not sure what to do here.

Suggestion is to ERR_set_mark in the start of the function and ERR_pop_to_mark at the end.